pkg/api/types.go

```diff
@@ -396,6 +398,24 @@ type NFSVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the lifetime of a pod
+type GlusterfsVolumeSource struct {
+	// Required: hosts are Glusterfs hosts address array
+	Hosts []string `json:"hosts"`
```

Comment:
This seems like not an ideal thing to require every user to put in a pod spec.  It requires specific knowledge of the cluster's nodes, which is something we would like to hide from the users where possible.  And, the set of Hosts might change quite a bit over time, making pod specs get out of date.

What are some other alternatives?
- require user to setup dynamic DNS and put that dns name here?
- setup a kubernetes service for the set of hosts (whether or not the hosts actually run in pods, this is possible).  Refer to that service here.  Kubelet pulls the service's endpoints at mount time.
- maintain a list of Hosts in a kubenetes `endpoints` object and refer to that here.  Kubelet can pull endpoints at mount time.